tests/mesolve/test_rouchon.py

```diff
@@ -1,48 +1,60 @@
-from math import pi
+import pytest
 
 import dynamiqs as dq
 
-from .me_solver_tester import MEAdjointSolverTester
-from .open_system import LeakyCavity
+from ..solver_tester import SolverTester
+from .open_system import grad_leaky_cavity_8, leaky_cavity_8
 
-leaky_cavity_8 = LeakyCavity(n=8, kappa=2 * pi, delta=2 * pi, alpha0=1.0)
-grad_leaky_cavity_8 = LeakyCavity(
-    n=8, kappa=2 * pi, delta=2 * pi, alpha0=1.0, requires_grad=True
-)
 
-
-class TestMERouchon1(MEAdjointSolverTester):
+class TestMERouchon1(SolverTester):
     def test_batching(self):
         options = dq.options.Rouchon1(dt=1e-2)
         self._test_batching(options, leaky_cavity_8)
 
-        options = dq.options.Rouchon1(dt=1e-2, sqrt_normalization=True)
-        self._test_batching(options, leaky_cavity_8)
-
-    def test_y_save(self):
-        options = dq.options.Rouchon1(dt=1e-3)
-        self._test_y_save(options, leaky_cavity_8, num_t_save=11)
-
-        options = dq.options.Rouchon1(dt=1e-3, sqrt_normalization=True)
-        self._test_y_save(options, leaky_cavity_8, num_t_save=11)
-
-    def test_adjoint(self):
-        options = dq.options.Rouchon1(dt=1e-3)
-        self._test_adjoint(options, grad_leaky_cavity_8, num_t_save=11)
-
-        options = dq.options.Rouchon1(dt=1e-3, sqrt_normalization=True)
-        self._test_adjoint(options, grad_leaky_cavity_8, num_t_save=11)
-
-
-class TestMERouchon2(MEAdjointSolverTester):
+    @pytest.mark.parametrize('sqrt_normalization', [False, True])
+    def test_correctness(self, sqrt_normalization):
+        options = dq.options.Rouchon1(dt=1e-3, sqrt_normalization=sqrt_normalization)
+        self._test_correctness(options, leaky_cavity_8, num_t_save=11)
+
+    @pytest.mark.parametrize('sqrt_normalization', [False, True])
+    def test_autograd(self, sqrt_normalization):
+        options = dq.options.Rouchon1(
+            dt=1e-3, sqrt_normalization=sqrt_normalization, gradient_alg='autograd'
+        )
+        self._test_gradient(
+            options, grad_leaky_cavity_8, num_t_save=11, rtol=1e-2, atol=1e-2
```

Comment:
Could this pass with lower `rtol` or `atol`? It seems quite high right now.

Reply:
(but don't reduce `dt` it's already good enough if the test can't do better)

Reply:
I lower all the rtol and atol to their min values in next commit 😉